public/video-ui/src/util/parseGridMetadata.js

```diff
@@ -10,22 +10,69 @@ function parseMimeType(mimeType) {
   return mimeType;
 }
 
-function parseAsset(asset) {
+function parseAsset(asset, aspectRatio) {
   return {
     file: asset.secureUrl,
     mimeType: parseMimeType(asset.mimeType),
     size: asset.size,
+    aspectRatio: aspectRatio,
     dimensions: {
       width: asset.dimensions.width,
-      height: asset.dimensions.height
+      height: asset.dimensions.height,
     }
   };
+};
+
+export function parseImageFromGridCrop(cropData, imageData) {
+  const aspectRatio = cropData.specification.aspectRatio;
+  return {
+    assets: cropData.assets.map((asset) => parseAsset(asset, aspectRatio)),
+    master: parseAsset(cropData.master, aspectRatio),
+    mediaId: cropData.specification.uri,
+    source: imageData.data.metadata.credit
+  };
 }
 
-export function parseImageFromGridCrop(cropData) {
+export function parseComposerDataFromImage(image, trail) {
+
+  const urlParts = image.mediaId.split('/');
+  const mediaId = urlParts[urlParts.length - 1];
```

Comment:
this is really similar to https://github.com/guardian/media-atom-maker/pull/534/files#diff-49e3809db1c472d315c7c0d48cccf491R67, could we abstract it into a shared function?